lisp/ess-inf.el

```diff
@@ -1293,43 +1293,44 @@ wrapping the code into:
-              (set-process-buffer proc out-buffer)
-              (set-process-filter proc 'inferior-ess-ordinary-filter)
-              (with-current-buffer out-buffer
-                (ess-setq-vars-local proc-alist)
-                (setq buffer-read-only nil)
-                (erase-buffer)
-                (set-marker (process-mark proc) (point-min))
-                (inferior-ess-mark-as-busy proc)
-                (process-send-string proc rich-cmd)
-                ;; Need time for ess-create-object-name-db on PC
-                (if no-prompt-check
-                    (sleep-for 0.02)   ; 0.1 is noticeable!
-                  (unless (ess-wait-for-process proc nil wait force-redisplay timeout)
-                    (error "Timeout during background ESS command `%s'"
-                           (ess--strip-final-newlines cmd)))
-                  ;; Remove prompt. If output is cat(..)ed without a
-                  ;; final newline, this deletes the last line of output.
+            (unwind-protect
+                (progn
+                  (when use-sentinel
+                    (process-put proc 'ess-output-sentinel sentinel))
+                  (set-process-buffer proc out-buffer)
+                  (set-process-filter proc 'inferior-ess-ordinary-filter)
+                  (with-current-buffer out-buffer
+                    (ess-setq-vars-local proc-alist)
+                    (setq buffer-read-only nil)
+                    (erase-buffer)
+                    (set-marker (process-mark proc) (point-min))
+                    (inferior-ess-mark-as-busy proc)
+                    (process-send-string proc rich-cmd)
+                    ;; Need time for ess-create-object-name-db on PC
+                    (if no-prompt-check
+                        (sleep-for 0.02) ; 0.1 is noticeable!
+                      (unless (ess-wait-for-process proc nil wait force-redisplay timeout)
+                        (error "Timeout during background ESS command `%s'"
+                               (ess--strip-final-newlines cmd)))
+                      ;; Remove prompt. If output is cat(..)ed without a
+                      ;; final newline, this deletes the last line of output.
+                      (goto-char (point-max))
+                      (delete-region (point-at-bol) (point-max))))
+                  (setq early-exit nil))
+              (when early-exit
+                (with-current-buffer out-buffer
+                  (goto-char (point-min))
+                  (when (and use-sentinel
+                             (not (re-search-forward
+                                   (inferior-ess--sentinel-start-re sentinel)
+                                   nil t)))
+                    ;; CMD probably failed to parse if the start sentinel
+                    ;; can't be found in the output. Disable the sentinel
+                    ;; before interrupt to avoid a freeze.
+                    (process-put proc 'ess-output-sentinel nil))
```

Comment:
proc variables need not actually be namespaced. In the entire codebase they are not.

Reply:
I noticed they were not but I think we should start namespacing them if ESS really intends to be generic.

Reply:
It's our own process, so we ca default to non-namespaced symbols. If other tooling use our process plist, they have to namespace.

Reply:
They are not necessarily our own process since we have `ess-remote`.

Reply:
ess-remote hijacks the process. There is no way back afterwards. 

Anyways, a tiny issue. I would keep small keywords for simplicity.

Reply:
There could also be minor modes interacting with the process? I agree it's a tiny issue and I don't mind removing the prefix.

Reply:
> There could also be minor modes interacting with the process?
Maybe, but those will use their own prefixes. 

>  I agree it's a tiny issue and I don't mind removing the prefix.

Well, the main thing is to keep it consistent. Either we keep all un-prefixed, or all prefixed. Changing now into prefixed seems an overkill of the work for little gain (little loss, actually, because it's uglier :).  Also who knows, maybe some dude/external pakage is using 'busy marker for something in their code. So I am afraid we are stuck with unprefixed keywords.

Reply:
s/dude/person. Actually the use of prefixes would allow us to use `ess--` for private internal state. Let's just consider that all these variables are private anyway. I'll rename to `output-sentinel` after we merge or close the PR.